test/libethereum/canonblockchain.cpp

```diff
@@ -0,0 +1,119 @@
+/*
+	This file is part of cpp-ethereum.
+
+	cpp-ethereum is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	cpp-ethereum is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** @file canonblockchain.cpp
+ * @author Dimitry Khokhlov <dimitry@ethdev.com>
+ * @date 2015
+ * CanonBlockChain test functions.
+ */
+
+#include <libethereum/CanonBlockChain.h>
+#include <libdevcore/TransientDirectory.h>
+#include <libethereum/Block.h>
+#include <test/TestHelper.h>
+
+using namespace std;
+using namespace dev;
+using namespace dev::eth;
+
+std::string const c_testGenesis = R"(
```

Comment:
this is for testing blockchain setGenesis()  function.
I have to know what values to expect for defined genesis. 

see report on this coverage:  ethereum/webthree-umbrella#74